tests/mvp-demo/mvp-demo.go

```diff
@@ -246,6 +254,31 @@ var _ = framework.MvpDemoSuiteDescribe("MVP Demo tests", Label("mvp-demo"), func
 			Expect(snapshot.Name).ToNot(BeEmpty())
 		})
 
+		It("Integration Test PipelineRun is created", func() {
+			Eventually(func() bool {
+				testPipelinerun, err = f.AsKubeAdmin.IntegrationController.GetIntegrationPipelineRun(integrationTestScenario.Name, snapshot.Name, userNamespace)
+				if err != nil {
+					GinkgoWriter.Printf("failed to get Integration test PipelineRun for a snapshot '%s' in '%s' namespace: %+v\n", snapshot.Name, userNamespace, err)
+					return false
+				}
+				return testPipelinerun.HasStarted()
+			}, pipelineRunStartedTimeout, defaultPollingInterval).Should(BeTrue())
+			Expect(testPipelinerun.Spec.PipelineRef.Bundle).To(ContainSubstring(integrationTestScenario.Spec.Bundle))
+			Expect(testPipelinerun.Labels["appstudio.openshift.io/snapshot"]).To(ContainSubstring(snapshot.Name))
+			Expect(testPipelinerun.Labels["test.appstudio.openshift.io/scenario"]).To(ContainSubstring(integrationTestScenario.Name))
+		})
+
+		It("Integration Test PipelineRun should eventually succeed", func() {
+			Eventually(func() bool {
```

Comment:
is this `Eventually` wrapper needed since the assertion below is executed after the pipelinerun is finished (i.e. succeeded/failed)?

Reply:
bump

Reply:
@psturc, I don't quite clear what assertions that are meant? Are they Line#277-279?Expect(testPipelinerun.Spec.PipelineRef.Bundle).To(ContainSubstring(integrationTestScenario.Spec.Bundle))
                        Expect(testPipelinerun.Labels["appstudio.openshift.io/snapshot"]).To(ContainSubstring(snapshot.Name))
                        Expect(testPipelinerun.Labels["test.appstudio.openshift.io/scenario"]).To(ContainSubstring(integrationTestScenario.Name))

Reply:
nope, I meant
```			Expect(f.AsKubeAdmin.IntegrationController.WaitForIntegrationPipelineToBeFinished(f.AsKubeAdmin.CommonController, integrationTestScenario, snapshot, appName, userNamespace)).To(Succeed(), "Error when waiting for a integration pipeline to finish")
```

Reply:
I think the Eventually in Line 272 is needed if there is any delay or issue and we always use Eventually when testing if the pipelinerun is started,  right?

Reply:
hmm we already checked that the pipelinerun exists [here](https://github.com/redhat-appstudio/e2e-tests/pull/512/files#diff-aa6a4487424b68bc1083984374ce6142929fcd35ce49f9175ce9a718fd1a59b9R270)

`WaitForIntegrationPipelineToBeFinished` method already [does polling](https://github.com/redhat-appstudio/e2e-tests/blob/5c5c2e57c96288200250adca4d7f691cfdb54767/pkg/utils/integration/controller.go#L344), so that's why I don't understand we are adding it to another polling wrapper (`Eventually`)

Reply:
Updated, thanks!